challenges/2021-04-13-single-and-ready-to-mingle/solutions/ruby/ericwburden/ericwburden.rb

```diff
@@ -0,0 +1,45 @@
+# # Single and Ready to Mingle
+# 
+# Given an array of integers in which two elements appear exactly once and all other
+# elements appear exactly twice, find the two elements that appear only once.
+# 
+# ## Business Rules/Errata
+# 
+# - You can assume that the input will **always** include at least two and exactly two
+#   elements that appear only once.
+# - You may not assume that the list will be sorted ahead of time.
+# - Extra Challenge: Can you complete this puzzle in linear time and constant space?
+# - Extra Extra Challenge: Can you make your solution generic over other input types?
+# 
+# ## Examples
+# 
+# ```
+# input = [2, 4, 6, 8, 10, 2, 6, 10];
+# find_singles(input)  // [4, 8]
+# ```
+# ```
+# input = [1, 2, 3, 2, 1, 4, 4, 6, 7, 8, 7, 6, 9, 9];
+# find_singles(input)  // [3, 8]
+# ```
+
+require 'set'
+
+def find_singles(numbers)
+  if !numbers.all? {|x| x.is_a?(Integer) } or !numbers.kind_of?(Array)
```

Comment:
Typically ruby "style" is to use truthy comparisons.

`!numbers.all? {|x| x.is_a?(Integer) }`

could be 

`numbers.none? { |x| x.is_a?(Integer) }`


Also, because of ruby's object inheritance there's a _bit_ of difference between `is_a?`/`kind_of` (which are synonymous) and the preferred `instance_of?` which won't allow subclasses

https://apidock.com/ruby/Object/instance_of%3F